client/src/components/Messages.jsx

```diff
@@ -8,20 +8,70 @@ import {
   List,
   Fab,
 } from '@material-ui/core';
+import { useLocation } from 'react-router-dom';
 import Message from './Message.jsx';
 import UserContacted from './UserContacted.jsx';
 import { AuthContext } from '../contexts/index.jsx';
 import dummyData from '../../../messagesDummyData';
 
-function Messages({ userId }) {
-  const [usersContacted, setUsersContacted] = useState(dummyData.usersContacted);
+function Messages() {
+  const [usersContacted, setUsersContacted] = useState([]);
   const [messages, setMessages] = useState([]);
-  const [userContactedId, setUserContactedId] = useState('null');
-  const { currentUser, addMessage, getMessages } = useContext(AuthContext);
+  const [userContactedId, setUserContactedId] = useState(null);
+  const {
+    currentUser,
+    addMessage,
+    getMessages,
+    getProfile,
+    updateUsersContacted,
+  } = useContext(AuthContext);
+  const didMounted = useRef(false);
+  const location = useLocation();
+  const profile = location.state;
 
-  // const getUsersContacted = () => {
-  //   return false;
-  // };
+  useEffect(() => {
+    if (didMounted.current) {
+      if (profile) {
+        let userAlreadyContacted = false;
+        for (let i = 0; i < usersContacted.length; i += 1) {
```

Comment:
You could consider replacing this for loop with a [some function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)